custom/src/main/java/co/elastic/otel/dynamicconfig/CentralConfig.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel.dynamicconfig;
+
+import co.elastic.opamp.client.CentralConfigurationManager;
+import co.elastic.opamp.client.CentralConfigurationProcessor;
+import co.elastic.otel.logging.AgentLog;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import java.text.MessageFormat;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public class CentralConfig {
+  private static final Logger logger = Logger.getLogger(CentralConfig.class.getName());
+
+  static {
+    DynamicConfigurationPropertyChecker.startCheckerThread();
+  }
+
+  public static void init(SdkTracerProviderBuilder providerBuilder, ConfigProperties properties) {
+    // TODO flip default when EDOT collector supports op amp
+    boolean startOpAmp = properties.getBoolean("elastic.otel.opamp.start", false);
+    if (!startOpAmp) {
+      return;
+    }
+    String serviceName = getServiceName(properties);
```

Comment:
This won't take service names derived from `ResourceProvider`s into account, will it?
If not, we should probably add a TODO here?

Reply:
I was asking because `getServiceName(properties)` only reads from the `ConfigurationProperties` and not potential changes from `ResourceProviders` which don't write those, but directly set the `Resource`, e.g. [AppServiceNameProvider](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/f4ff5f4b02df0b1572233ca9bfdd124489cd3b53/resource-providers/src/main/java/io/opentelemetry/contrib/resourceproviders/AppServerServiceNameProvider.java#L24).

But maybe there is some magic behind the curtains making this work? So if you tested it this is fine, otherwise we should add a TODO